Login and signup pages styled to match mockups

simalicrum opened the pull request:
### What this PR does (required):
- Login and signup pages styled to match mockups

### Screenshots / Videos (front-end only):
![image](https://user-images.githubusercontent.com/15878731/138609516-064b9fcc-cfd8-46d3-930a-71b318686cd7.png)

### Any information needed to test this feature (required):
- working mongodb required

rajivtitus commented on client/src/components/DemoButton/DemoButton.tsx:
```diff
@@ -0,0 +1,53 @@
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { CircularProgress } from '@material-ui/core';
+
+const demoUser = { email: 'user@demo.com', password: 'password' };
```
👍🏽

rajivtitus replied:
I think this feature was already a part of a previous PR which has been approved? 🤔 

When you have some time to spare, I would highly recommend creating a new repo and practice creating PRs - branching off of other branches - rebasing branches - et cetera.

rajivtitus commented on client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:
```diff
@@ -40,7 +41,7 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
       }}
       validationSchema={Yup.object().shape({
-        username: Yup.string().required('Username is required').max(40, 'Username is too long'),
+        username: Yup.string().required('Name is required').max(40, 'Name is too long'),
```
```suggestion
        name: Yup.string().required('Name is required').max(40, 'Name is too long'),
```

rajivtitus commented on client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:
```diff
@@ -40,7 +41,7 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
         username: '',
```
```suggestion
        name: '',
```

rajivtitus commented on client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:
```diff
@@ -51,66 +52,60 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
     >
       {({ handleSubmit, handleChange, values, touched, errors, isSubmitting }) => (
         <form onSubmit={handleSubmit} className={classes.form} noValidate>
+          <label htmlFor="username">
```
```suggestion
          <label htmlFor="name">
```

Ditto throughout!

simalicrum replied:
For consistency I've changed the text field and label to 'Username' and placeholder to 'Create a username' then username is disaggregated from first and last name which is more typical of login interfaces.

rajivtitus reviewed (approved):
Nice work! There are some minor suggestions, please address them prior to merging. Thanks 👍🏽

simalicrum commented:
I've moved the shared UI elements from the login and signup into ModalButton and ModalPaper function components for easy of styling and future reuse. Also I fixed a couple of outstanding CSS issues.
